### PR TITLE
Allow configuration of the number of max requests handled per puppet …

### PR DESCRIPTION
**Pull request by dfairhurst**
…master process

### CODE REVIEW

**Comment by dfairhurst** [comment]
In order to work around https://tickets.puppetlabs.com/browse/PUP-731 we require to be able to be able to customise the number of requests handled per puppet master process invocation (to 1 per process).  This change makes the value customisable.
In future, it could make sense to make all these kinds of parameters customisable, but this is the one we require currently.

**Comment by dfairhurst** [comment]
Any chance to get this merged?
Thanks :)

File: templates/puppetmaster-vhost.conf.erb [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 PassengerUseGlobalQueue On
 
-# Recycle master processes after they service 1000 requests
-PassengerMaxRequests 1000
+# Recycle master processes after they service <%= @passenger_max_requests %> requests
```
Comment (ghoneycutt) [inline templates/puppetmaster-vhost.conf.erb]: Would recommend changing to

```
# Recycle master processes after they service the number of requests specified by PassengerMaxRequests. Default is 1000.
```

**Comment by dfairhurst** [comment]
Updated as suggested

**Comment by ghoneycutt** [comment]
Apologies, one more thing I had previously forgotten. Each parameter is documented in the README. Could you please document this new parameter in the style of the others.

**Comment by dfairhurst** [comment]
Well spotted - didn't think of that.  README now updated

**Comment by aschaber1** [comment]
@ghoneycutt is there anything else to be done, for this to get merged?

**Comment by ghoneycutt** [comment]
Thank you!!

Released in v2.18.0

**Comment by dfairhurst** [comment]
Many thanks

**Pull request merged**